internal/services/tesla_api_service.go

```diff
@@ -0,0 +1,129 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/DIMO-Network/devices-api/internal/config"
+	"github.com/rs/zerolog"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+//go:generate mockgen -source tesla_api_service.go -destination mocks/tesla_api_service_mock.go
+type TeslaAPIService interface {
+	CompleteTeslaAuthCodeExchange(authCode, redirectURI, region string) (*TeslaAuthCodeResponse, error)
+	GetVehicles(token, region string) ([]TeslaVehicle, error)
+}
+
+type teslaAPIService struct {
+	Settings   *config.Settings
+	HTTPClient *http.Client
+	log        *zerolog.Logger
+}
+
+func NewTeslaAPIService(settings *config.Settings, logger *zerolog.Logger) TeslaAPIService {
+	return &teslaAPIService{
+		Settings: settings,
+		HTTPClient: &http.Client{
+			Timeout: 5 * time.Second,
+		},
+		log: logger,
+	}
+}
+
+func (t *teslaAPIService) CompleteTeslaAuthCodeExchange(authCode, redirectURI, region string) (*TeslaAuthCodeResponse, error) {
+	u, err := url.Parse(t.Settings.TeslaAuthorization.TokenAuthURL)
```

Comment:
Did you see the oauth2 code I posted? Does that read better to you at all?

Reply:
Let me remind

https://github.com/DIMO-Network/testla/blob/799589ea9829c2aed5aedc741aad38575f7c98a3/auth.go#L49